src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -504,15 +502,10 @@ private ExprCall BindToMethod(MethWithInst mwi, Expr pArguments, ExprMemberGroup
             pObject = AdjustMemberObject(mwi, pObject, out fConstrained);
             pMemGroup.OptionalObject = pObject;
 
-            CType pReturnType = null;
-            if ((flags & (MemLookFlags.Ctor | MemLookFlags.NewObj)) == (MemLookFlags.Ctor | MemLookFlags.NewObj))
-            {
-                pReturnType = mwi.Ats;
-            }
-            else
-            {
-                pReturnType = GetTypes().SubstType(mwi.Meth().RetType, mwi.GetType(), mwi.TypeArgs);
-            }
+            CType pReturnType =
```

Comment:
I'd suggest keeping the old if statement (with removing `= null`) for readability.

Reply:
I think shortening the test part of the conditional makes it more readable, but readability is subjective and I've a high tolerance for conditionals over if/else (blame coding javascript in the days when most customers where on dialup, minifiers weren't common, and managers set small kilobyte limits) so if you'd still prefer the if/else I'll go with that.

Reply:
Scratch that. That's not equivalent at all. Turning back into an if/else.